packages/datadog_flutter_plugin/lib/src/rum/rum_gesture_detector.dart

```diff
@@ -0,0 +1,272 @@
+/// which it will use for the description of the action. In other cases, it will
+/// look for a child [Semantics] object, or an [Icon] with its [semanticsLabel]
+/// property set.
+///
+/// Alternately, you can enclose any Widget tree with a
+/// [RumUserActionAnnotation], which will use the provided description when
+/// reporting user actions detected in the child tree, without changing the
+/// Semantics of the tree.
+class RumUserActionDetector extends StatefulWidget {
+  @internal
+  static final elementMap = <RumUserActionDetector, Element>{};
+
+  /// The instance of RUM to report to.
+  final DdRum? rum;
+
+  /// The Widget tree to detect gestures in.
+  final Widget child;
+
+  const RumUserActionDetector({
+    Key? key,
+    required this.rum,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  StatefulElement createElement() {
+    final e = super.createElement();
+    elementMap[this] = e;
+    return e;
+  }
+
+  @override
+  State<RumUserActionDetector> createState() => _RumUserActionDetectorState();
+}
+
+class _RumUserActionDetectorState extends State<RumUserActionDetector> {
+  final _listenerKey = GlobalKey();
+
+  int? _lastPointerId;
+  Offset? _lastPointerDownLocation;
+
+  @override
+  void didUpdateWidget(covariant RumUserActionDetector oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    var element = RumUserActionDetector.elementMap[oldWidget];
+    if (element != null) {
+      RumUserActionDetector.elementMap.remove(oldWidget);
+      RumUserActionDetector.elementMap[widget] = element;
+    } else {
+      // Telemetry -- this shouldn't happen
+    }
+  }
+
+  @override
+  void dispose() {
+    RumUserActionDetector.elementMap.remove(widget);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Listener(
+      key: _listenerKey,
+      behavior: HitTestBehavior.translucent,
+      onPointerDown: _onPointerDown,
+      onPointerUp: _onPointerUp,
+      child: widget.child,
+    );
+  }
+
+  void _onPointerDown(PointerDownEvent event) {
+    _lastPointerId = event.pointer;
+    _lastPointerDownLocation = event.localPosition;
+  }
+
+  void _onPointerUp(PointerUpEvent event) {
+    if (_lastPointerDownLocation != null && event.pointer == _lastPointerId) {
+      final distanceOffset = Offset(
+          _lastPointerDownLocation!.dx - event.localPosition.dx,
+          _lastPointerDownLocation!.dy - event.localPosition.dy);
+
+      final distanceSquared = distanceOffset.distanceSquared;
+      if (distanceSquared < _tapSlopSquared) {
+        _onPerformActionAt(event.localPosition, RumUserActionType.tap);
+      }
+    }
+  }
+
+  void _onPerformActionAt(Offset position, RumUserActionType action) {
+    final elementDescription = _getDetectingElementAtPosition(position);
+
+    if (elementDescription != null) {
+      widget.rum?.addUserAction(
+          RumUserActionType.tap, elementDescription.elementDescription);
```

Comment:
`action` argument should be used here

```suggestion
          action, elementDescription.elementDescription);
```